### ISSUE TITLE
Time issue: Exception: AttributeError: module 'azure.functions' has no attribute 'get_binding_registry'

### ISSUE THREAD

**Issue by docgregt** [Azure/azure-functions-core-tools#1455]
Firstly, this code has been running for months. Secondly, the last time I deployed two days ago the deployment worked fine. Now today I deploy my function app that two timers in it and neither timer will work.
I get following errors:
Result: Failure
Exception: AttributeError: module 'azure.functions' has no attribute 'get_binding_registry'
Stack:   File "/usr/local/lib/python3.6/site-packages/azure/functions_worker/dispatcher.py", line 241, in _handle__function_load_request
    function_id, func, func_request.metadata)
  File "/usr/local/lib/python3.6/site-packages/azure/functions_worker/functions.py", line 194, in add_function
    param_bind_type, param_py_type)
  File "/usr/local/lib/python3.6/site-packages/azure/functions_worker/bindings/meta.py", line 36, in check_input_type_annotation
    binding = get_binding(binding)
  File "/usr/local/lib/python3.6/site-packages/azure/functions_worker/bindings/meta.py", line 20, in get_binding
    registry = get_binding_registry()
  File "/usr/local/lib/python3.6/site-packages/azure/functions_worker/bindings/meta.py", line 13, in get_binding_registry
    return func.get_binding_registry()

I tried rebuilding everything from scratch. Deleted my .env, etc. to no avail. Deleted the app service and recreated it. Nothing helps.

The timers work fine in Visual Studio Code running on my local machine.
Python 3.6.8
CLI 2.0.67
azure-functions==1.0.0a5
azure-functions-worker==1.0.0a6

I dont know what else to do. I would appreciate how to solve it.
Greg

**Comment by mebibou** [comment]
I am just seing the same error on my function that worked fine before. For me it happens on a HTTP function, and sometimes I have the issue sometimes not, very frustrating.
The app is deployed under the Service plan `EastUSLinuxDynamicPlan (Consumption)`

So recently I couldn't deploy to the region, now the apps are failing. Could you tell us if it is ready or we should move away from Azure function?

**Comment by mkolpin** [comment]
I can confirm this issue for all timer trigger functions. Python Azure Functions are completely unusable currently. Redeploying to a freshly set up function app does not change this behaviour.
Even the minimal timer trigger example created in VS Code throws the same exception.

**Comment by mkolpin** [comment]
I can confirm this issue for all timer trigger functions. Python Azure Functions are completely unusable currently. Redeploying to a freshly set up function app does not change this behaviour.
Even the minimal timer trigger example created in VS Code throws the same exception.

To add to this: the same error ocurrs for previously created functions since about a week ago. So the issue seems to be related to the runtime itself and not an update to the core tools.

**Comment by docgregt** [comment]
Breaking is understatement. I removed the two lines from my requirements.txt and now no function works at all. Forget timer. None of them work.
![image](https://user-images.githubusercontent.com/43346207/61893214-c9932000-aeca-11e9-82a1-887e7c8cb742.png)
In case you cant see the image it says 'function runtime is unable to start'.

**Comment by mebibou** [comment]
@mkolpin yes it fixed my issue, thanks!

**Comment by docgregt** [comment]
The #496 is not a solution. It breaks everything. I put these two lines back into requirements.txt:
azure-functions==1.0.0b5
azure-functions-worker==1.0.0b10

Then I switched from consumption (which might work now) to standard Linux App Service (which i already had for other things) and now everything works for me. I do not consider this issue closed.
MS has problems and they need to explain to us the proper way to set things up.

**Comment by mebibou** [comment]
@maiqbal11 Could you help out @docgregt here?
imho, just putting an issue on github stating "will need to remove them to avoid breaking their app" is very shocking. I mean I literally had to see my apps in production not working anymore to search on github, find a related issue, then apply the fix, re-deploy... I mean can't you guys first put a warning somewhere? or send a newsletter to all accounts who are using the function apps to warn them about the upcoming changes? for me, this is very unprofessional

**Comment by docgregt** [comment]
FYI. I think there are compatibility issues with:
cli
core-tools
cosmos db sdk

I am using all three. I see different issues being reported. Perhaps the three use different supporting libs. Here is a related case which helped me:
https://github.com/Azure/azure-functions-python-worker/issues/442
But for sure https://github.com/Azure/azure-functions-python-worker/issues/496 is not good and breaks everything.

**Comment by maiqbal11** [comment]
@docgregt, what version of core tools are you using?

**Comment by docgregt** [comment]
![image](https://user-images.githubusercontent.com/43346207/61966882-1f320000-af91-11e9-84dc-571d9370c54e.png)

is this what you are looking for?

**Comment by maiqbal11** [comment]
Yes, that's it. Are you able to run the function locally?

**Comment by docgregt** [comment]
Yes, all my functions have run for many months. The timers stopped working on July 24th when I deployed a fix to my code. And they work now both locally and deployed as I mentioned above. I do not know if they work on consumption plan as I moved away from that. Seems to be a difference between Consumption and App Service plans.
Removing azure-functions and azure-functions-worker from requirements.txt broke all functions. I put them back in and now everything works in App Service plan.

**Comment by maiqbal11** [comment]
Would you be able to share the name of your function app?

**Comment by docgregt** [comment]
i4pyservices

**Comment by docgregt** [comment]
fyi, this occurred on the 24th:
![image](https://user-images.githubusercontent.com/43346207/61968433-41c61800-af95-11e9-8fd0-6f7c50a006b8.png)

At that point I was using Consumption service.

**Comment by maiqbal11** [comment]
@docgregt, do you have AlwaysOn enabled for your App Service Function App?

**Comment by docgregt** [comment]
![image](https://user-images.githubusercontent.com/43346207/61973542-78099480-afa1-11e9-8204-2f751234c505.png)

**Comment by maiqbal11** [comment]
@docgregt, could you also share your `requirements.txt`?

**Comment by docgregt** [comment]
adal==1.2.1
antlr4-python3-runtime==4.7.2
applicationinsights==0.11.7
argcomplete==1.10.0
asn1crypto==0.24.0
astroid==2.1.0
azure-batch==6.0.0
azure-cli==2.0.67
azure-common==1.1.23
azure-cosmos==3.0.2
azure-functions==1.0.0b5
azure-functions-worker==1.0.0b10
azure-functions-devops-build==0.0.22
azure-graphrbac==0.60.0
azure-keyvault==1.1.0
azure-mgmt-authorization==0.50.0
azure-mgmt-dns==2.1.0
bcrypt==3.1.7
certifi==2018.11.29
cffi==1.12.3
cftime==1.0.3.4
chardet==3.0.4
colorama==0.4.1
cryptography==2.7
DateTimeRange==0.5.5
fabric==2.4.0
geographiclib==1.49
grpcio==1.14.2
grpcio-tools==1.14.2
html2text==2018.1.9
humanfriendly==4.18
idna==2.8
invoke==1.2.0
ipaddress==1.0.22
isodate==0.6.0
isort==4.3.4
Jinja2==2.10.1
jmespath==0.9.4
knack==0.6.2
lazy-object-proxy==1.3.1
mail-parser==3.8.1
MarkupSafe==1.1.1
mbstrdecoder==0.7.0
mccabe==0.6.1
mock==3.0.5
msrest==0.6.8
msrestazure==0.6.1
netCDF4==1.4.2
numpy==1.16.0
oauthlib==3.0.1
pandas==0.24.0
paramiko==2.6.0
portalocker==1.2.1
prompt-toolkit==1.0.16
protobuf==3.6.1
psutil==5.6.3
ptvsd==4.2.2
pycparser==2.19
Pygments==2.4.2
PyJWT==1.7.1
pylint==2.2.2
PyNaCl==1.3.0
pyOpenSSL==19.0.0
pyperclip==1.7.0
pyreadline==2.1
python-dateutil==2.8.0
pytz==2018.9
PyYAML==5.1.1
requests==2.21.0
requests-oauthlib==1.2.0
scipy==1.2.0
scp==0.13.2
simplejson==3.16.0
six==1.12.0
sshtunnel==0.1.5
tabulate==0.8.3
typed-ast==1.2.0
typepy==0.4.0
urllib3==1.24.1
vsts==0.1.25
vsts-cd-manager==1.0.2
wcwidth==0.1.7
websocket-client==0.56.0
wrapt==1.11.1
xarray==0.11.3
xlrd==1.2.0
xmltodict==0.12.0

**Comment by maiqbal11** [comment]
> FYI. I think there are compatibility issues with:
> cli
> core-tools
> cosmos db sdk
> 
> I am using all three. I see different issues being reported. Perhaps the three use different supporting libs. Here is a related case which helped me:

@docgregt, there are definitely some compatibility issues that we are seeing on our end between `azure-cosmos` and `azure-functions-worker`. These are related to some `azure` packages declaring explicit namespaces and others not. This also manifests differently between Consumption and App Service. We are looking into a fix and will update accordingly.

**Comment by docgregt** [comment]
Glad I was a bit helpful. Let me know when I can test it.

**Comment by maiqbal11** [comment]
@docgregt, greatly appreciate you reporting this! For anyone else who is experiencing issues on App Service, I'll just make an explicit note that they should include `azure-functions` and `azure-functions-worker` in their `requirements.txt`. I'll put this into our announcement as well for now as we work towards making the experience consistent across the platforms.
